src/pages/index.js

```diff
@@ -0,0 +1,312 @@
+  evt.preventDefault();
+  const submitButton = deleteForm.querySelector(".modal__button_type_delete");
+  submitButton.textContent = "Deleting...";
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteCardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Yes"; // Reset button text
+    });
+};
+
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
+// Create card element
+function getCardElement(data, userId) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  cardElement.id = `card-${Date.now()}`;
+
+  const cardElementDescription =
+    cardElement.querySelector(".card__description");
+  const cardElementImage = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardElementDescription.textContent = data.name;
+  cardElementImage.src = data.link;
+  cardElementImage.alt = data.name;
+
+  cardLikeButton.addEventListener("click", () => {
+    const isLiked = cardLikeButton.classList.contains(
+      "card__like-button_liked"
+    );
+    const likeAction = isLiked ? api.deleteLike : api.addLike;
+
+    likeAction
+      .call(api, data._id)
+      .then((updatedCard) => {
+        cardLikeButton.classList.toggle("card__like-button_liked");
+      })
+      .catch(console.error);
+  });
+
+  cardElementImage.addEventListener("click", () => {
+    previewImage.src = cardElementImage.src;
+    previewImage.alt = cardElementImage.alt;
+    previewCaption.textContent = cardElementImage.alt;
+    openModal(previewModal);
+  });
+
+  cardDeleteButton.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data);
+  });
+
+  return cardElement;
+}
+
+// Handle profile form
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const profileEditModal = document.querySelector("#profile__edit-modal");
+  const submitButton = profileEditModal.querySelector(".modal__submit-btn");
+  submitButton.textContent = "Save...";
+  api
+    .editUserInfo({
+      name: profileEditModalNameInput.value,
+      about: profileEditModalDescriptionInput.value,
+    })
+    .then(() => {
+      profileName.textContent = profileEditModalNameInput.value;
+      profileDescription.textContent = profileEditModalDescriptionInput.value;
+      closeModal(profileEditModal);
+      avatarForm.reset();
+      disableButton(submitButton, validationConfig);
+    })
+    .catch((err) => console.error(`Error: ${err}`))
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
+}
+
+// Handle new post form
+function handleNewPostFormSubmit(evt) {
+  evt.preventDefault();
+  const newPostModal = document.querySelector("#new__post-modal");
```

Comment:
`  const newPostModal = document.querySelector("#new__post-modal");
`

You've alredy found this modal in line 46